### ISSUE TITLE
Feature Requests

### ISSUE THREAD

**Issue by FluffyGhoster** [KAT-Advanced-Medical/KAM#29]
**A port of the feature requests from the Katalam/KAM repository and new additions from players**

**Description:**

Add a detailed description of the feature you wish, if it's deemed accepted it will be moved to a feature request by itself and a target version will be added, on the completed development a pull request will contain the Issue ID and the issue will be closed

**Comment by FluffyGhoster** [comment]
**Description:**
Flight considerations for MEDEVAC/CASEAVAC; atmospheric pressure changes for pneumothorax. Need to ensure that the patient is stable enough for transport via aviation (helicopter).

**How to implement:**
ACE already has an option to effect ballistics of a bullet via barometric pressure so there should be a way to check for altitude changes to effect the overall barometric pressure for the wounded as well.

----

[XSTAT ](https://www.revmedx.com/)- Injectable sponge type pellet system to keep massive blood loss from happening with a patient either with gunshot would or from amputation (evulsion) from IED's, wrecks, etc... The system creates a pressure to close off severed vessels (veins and arteries) to reduce/stop blood loss.

----

**Description:**
I reached out to you on discord but figured I'd post it here to make it more official. The Idea of having to have different size IV catheters/IOs and starting a "line" before being able to push fluids or medications into the patient. The limb the IV or IO is in is then the only limb that fluids can be pushed into. Be able to start multiple IV's on a patient. The size of the catheter can then determine the fluid transfusion rate etc.

**Steps to treat:**
ACE medical already has the option to determine transfusion rates in the settings. Should hopefully be simple to link the size of the catheter to the transfusion rate.

IV:
select limb, option to start (14, 16, 18, 20, 22, 24) gauge IV.
Once IV is in place give some type of indicator on that limb that an IV is established, same way with how there is an airway device notification.
Then in order to push fluids the provider has to select a limb that has an established IV (this could also allow the option to expand into dosed medications instead of the predosed auto injectors)

Also the possibility for an IV attempt to be unsuccessful and have to try again on the same or different limb.

IO:
An IO would be started the same way as an IV but should be faster and more reliable/secure since IRL it normally is.
Same steps as before but IO's have less size selections normally. (16, 18)
IO's should have some drawback for use on conscious patients like inflicting some pain if the patient is conscious. This will hopefully add some thought into the decision of and IO or IV on a patient.

Once an IV or IO is established on a patient is should stay with them until they are PAK'd or fully healed.

----

It would be nice to see chest seals being used for any sucking chest wound and the implementation of decompression needle(s) to replace the functionality of chest seals. I think this is a very simple addition that would add to the immersion of TCCC (Tactical Combat Casualty Care) and most medics to this day carry a decompression needle for this very reason (pneumothorax). Thanks

----

**Description:**

Add the ability to hang different types and sizes of IV solution on the IV stand. Saline, Blood, Plasma, Blood types in 250ml, 500ml, 1000ml...

**If possible add a idea to implement it (logical):**
Possibly use the same model for the IV stand if the smaller variants don't exist? Model can continue to be saline as long as the correct fluid is on the stand.

----

**Finished to copy over the Katalam/KAM repository Feature Request thread

**Comment by FluffyGhoster** [comment]
**Description:**
Reduce asystolic monitoring volume when you are near, it tend to pierce your ears overtime

**Implementation:**
I see two options: either just reduce the maximum volume, or mute it when the medic is near after a short time (like the medic press the "silence alarm" button)

**Comment by YetheSamartaka** [comment]
> 
> 
> **Description:**
> Reduce asystolic monitoring volume when you are near, it tend to pierce your ears overtime
> 
> **Implementation:**
> I see two options: either just reduce the maximum volume, or mute it when the medic is near after a short time (like the medic press the "silence alarm" button)

There is an option to mute the monitoring sound

**Comment by FluffyGhoster** [comment]
**Description:**
AED should apply shock pain only to people directly touching the patient, not just standing nearby

**Implementation:**
Currently the script search for people around the patient and give them ACE pain, however when a shock is administered only the people in direct contact to the patient risk to be affected, not just bystanders, hence the script should:

1) Select the people in the ACE treatment range
2) Check if they are doing a treatment to the specific patient
3) Apply pain to them if they are doing a treatment to the patient who is receiving the shock, with a small probability to cause an arrhythmia or stop the heart of the person that is touching the patient, possibly the probability depends on where the treatment is being applied (touching the torso is the maximum risk, touching the legs and arms is the lowest)

**Comment by FluffyGhoster** [comment]
> There is an option to mute the monitoring sound

I may have missed it then, where is it?

**Comment by YetheSamartaka** [comment]
> 
> 
> > There is an option to mute the monitoring sound
> 
> I may have missed it then, where is it?

ace interaction - turn down AED-X volume

**Comment by FluffyGhoster** [comment]
> > > There is an option to mute the monitoring sound
> > 
> > 
> > I may have missed it then, where is it?
> 
> ace interaction - turn down AED-X volume

Uhm no I don't see it, I open the patient, click "Monitor Vitals", you get the sound, but there's no option to mute it, either in the medical menu or on the ace interaction?

**Comment by FluffyGhoster** [comment]
**Description:**
AED is an automatic external defibrillator, so you shouldn't be able to shock the patient whenever he's on the ground, it should shock automatically on certain kind or arrythmic complexes, but not on medic's will/decision.
The AED should just be attached to the patient like the Monitor Vitals, then it will start the analysis of the rythm, if the rythm is shockable the AED will inform people to clear the patient, then perform the shock, and then go back in monitoring for a bit of time (I think it's 3 minutes?), the medic have no option to choose by himself if the AED will shock or not, otherwise it would be a manual defibrillator.
Also, the AED do not shock if the rythm is absend (asystolic), the protocol is to NOT shock asystolic patients but do CPR instead, you can think to a defibrillator as a resetter, it's not a jump starter, you don't use it if there's no electrical activity.

**Comment by Conrimaceogain** [comment]
Regarding decompression, I've been putting a little bit of work into implementing hemopneumorthox, so I might look to decompression too.

**Comment by Conrimaceogain** [comment]
I might also recommend splitting up these feature suggestions into separate issues to make tracking them and their acceptance/progress simpler.

**Comment by FluffyGhoster** [comment]
Would you prefer to have an enhancement each for every idea?
I thought going with the ACE3 style (a single thread with all the ideas, once accepted move them in a dedicated issue) was a nice idea, but please let me know if you prefer to just have an issue for each idea and I'll split them all

**Comment by Conrimaceogain** [comment]
Hmm, hadn't thought about it that way. That may work, let's give it a shot.

**Comment by FluffyGhoster** [comment]
**Description:**
Intubation is performed with a sedation therapy consisting of an agent (like Etomidate), we should provide such a medication and punish intubations that are done without it (chance of intubation not being applied? more desaturation for the patient? straight inability to intubate?)

This does not apply to nasal intubation and does not apply to patient in cardiac arrest which are unable to have such a reflex, but once they are responsive again we should apply a penality to them too (coughing with decreasing saturation?)

**Comment by FluffyGhoster** [comment]
**Description:**
Ability to remove intubation once it's not needed

----

**Description:**
Ambu baloon with the action to push it, will increase saturation when the push is performed if there's nothing obstructing the mouth, otherwise will exponentially reduce the saturation until sucking or, if done too much, until tracheotomy with suction is performed, would be cool to allow the connection to the ambu to the intubation or the tracheotomy mask

----
**Description:**
Manual defibrillator in the medical facilites, we can add a certain chance of the AED to not function properly and that would require the patient to be transported to the medical facility where the manual defibrillator is available (and only usable by doctors by default) that would then check the patient and shock it on his will.
This would couple very well with the feature below

----
**Description:**
A multiparametric screen able to be attached to the patient with a UI on it, it will then display the status of the patient with the SpO2, the SpCO2, and the rhythm (a picture that we can change, as generating it mathematically sounds like a nightmare, we can make a collection of PAAs for the conditions we want to show), this would couple well with a manual defibrillator because we can create our own statuses on the patient like a ventricular fibrillation, show the image on the screen that the doctor can then compare against a card (or not, and they do it the hard real way) and decide to shock it or use other medications (like amiodarone) to cardiovert the patient.
We should have a settings that allow the users to choose if they want this feature enabled or have the simple way, in case of the latter the AED will always work and it will cardiovert every condition.

**Comment by Katskan** [comment]
**Description:**
External jugular vein access. A separate option that may be restricted to high level personnel to establish access of the external jugular veins in the neck. Would mean an IV can be maintained and used even when all limbs require tourniquets.

**Comment by Katskan** [comment]
**Description:**
Manikin function to select specific number, location and type of injuries and airway conditions. Would aid in efficiency with training.

**Description:**
Switch Guedel->NPA (nasopharyngeal airway), the advantage of the NPA is that it can be used on patients with a gag reflex, and is often chosen over the guedel/OPA because of that. It may additionally be worth switching the King LT to an ET (endotracheal) tube, or adding an ET tube. ETT intubation is commonly done in military settings by higher level personnel like flight medics because prolonged use of supraglottic devices like the King can cause tissue necrosis in the esophagus.

**Comment by Yoshod** [comment]
**Description:**
This might be a bit much, but more realistic cardiac arrests would be nice. Currently as it stands all patients who are in cardiac arrest are asystole, which irl is not shockable. That would require manual CPR, epinephrine, and breathing management to get an output. It would be more accurate (as far as I understand it as somebody who is not medically trained) to include ventricular fibrillation (which can be caused by major trauma) or pulseless ventricular tachycardia. The latter would be simulated by not getting a pulse when checking on the neck or hands, but getting a tachycardic output on the AED X Series. I think what would make this difficult to implement would be only getting the audio of the ECG as compared to a visual ECG.

I will note I'm not a medical professional but friends and family of quite a few so most of this is what I've picked up from hearing them talk, asking questions, and my own research. If I'm getting stuff wrong then please let me know as I'd be interested to learn what's correct.

Edit:
Advised by a friend to mention another type of arrhythmia that could be included, pulseless electrical activity. Generally that would be treated with CPR and adrenaline (not defibrillation afaik) and would be sorted with treating the underlying cause of the PEA (such as major haemorrhage). Like the pulseless ventricular tachycardia this would be diagnosed by having electrical output on the monitor but no detectable pulse.

**Comment by Will-Nichols** [comment]
> 
> 
> **Description:**
> Manual defibrillator in the medical facilities, we can add a certain chance of the AED to not function properly and that would require the patient to be transported to the medical facility where the manual defibrillator is available (and only usable by doctors by default) that would then check the patient and shock it on his will.
> This would couple very well with the feature below
> 
> **Description:**
> A multi-parametric screen able to be attached to the patient with a UI on it, it will then display the status of the patient with the SpO2, the SpCO2, and the rhythm (a picture that we can change, as generating it mathematically sounds like a nightmare, we can make a collection of PAAs for the conditions we want to show), this would couple well with a manual defibrillator because we can create our own statuses on the patient like a ventricular fibrillation, show the image on the screen that the doctor can then compare against a card (or not, and they do it the hard real way) and decide to shock it or use other medications (like amiodarone) to cardio-vert the patient.
> We should have a settings that allow the users to choose if they want this feature enabled or have the simple way, in case of the latter the AED will always work and it will cardio-vert every condition.

Personally as a mission maker I don't want to have anything extra for my medics to have to carry into the field than they absolutely need. We are already loading them down with close to 100lb of gear including weapons, ammo, radios, medical stuff and we are having to config backpacks in ridiculous ways to make crap fit.

If I have them take an AED out with them then it needs to function more than 90% of the time. However anything else is just a reason to call for a medevac to get the casualty out of the field. All these extra jump starters need to be used in a "hospital" setting and not in the field. Once you get to connecting a patient to an AED in real life you need to be ready to move on to the next level of care.

If you are going to put these ultra-complex monitoring devices into the mod then certainly make them have a usability option for only in the hospital or medical vehicles. I don't believe they have a place "out in the field" attached to someone you are still putting white stuff on red stuff with and trying to stabilize.

**Comment by Katskan** [comment]
A loud, frequent gasping sound effect for pneumothorax or making it turn the torso red/another color would be very helpful, I have lots of grug brain infantry who are having a tough time with not noticing the pneumothorax because it requires reading, and it has not been drawn in colorful crayon and is therefore invisible to them

**Comment by Conrimaceogain** [comment]
> A loud, frequent gasping sound effect for pneumothorax or making it turn the torso red/another color would be very helpful, I have lots of grug brain infantry who are having a tough time with not noticing the pneumothorax because it requires reading, and it has not been drawn in colorful crayon and is therefore invisible to them

I snorted when I read the end of that sentence. Adding more indicators of the patient's condition is definitely something I'd like to do.

**Comment by YetheSamartaka** [comment]
> A loud, frequent gasping sound effect for pneumothorax or making it turn the torso red/another color would be very helpful, I have lots of grug brain infantry who are having a tough time with not noticing the pneumothorax because it requires reading, and it has not been drawn in colorful crayon and is therefore invisible to them

Problem with pneumothorax is, that in quite a lot of cases, patient won't make any sounds, in fact when someone has pneumothorax and you will use stethoscope to listen to their lungs, doctors recoqnize it that it makes less sound than usual. I'm not an medical expert, I just tried to make some research, so feel free to correct me.

From gameplay point of view, yes, we are going to add some graphical indication to the medical menu, but we are also considering the function, that you would have to "check if patient has pneumothorax/hemothorax or whatever". If anything such that will be added, don't worry, it will have on/off button so each community can decide how it will work for them.

**Comment by Katskan** [comment]
> > A loud, frequent gasping sound effect for pneumothorax or making it turn the torso red/another color would be very helpful, I have lots of grug brain infantry who are having a tough time with not noticing the pneumothorax because it requires reading, and it has not been drawn in colorful crayon and is therefore invisible to them
> 
> Problem with pneumothorax is, that in quite a lot of cases, patient won't make any sounds, in fact when someone has pneumothorax and you will use stethoscope to listen to their lungs, doctors recoqnize it that it makes less sound than usual. I'm not an medical expert, I just tried to make some research, so feel free to correct me.
> 
> From gameplay point of view, yes, we are going to add some graphical indication to the medical menu, but we are also considering the function, that you would have to "check if patient has pneumothorax/hemothorax or whatever". If anything such that will be added, don't worry, it will have on/off button so each community can decide how it will work for them.

Yeah in real world practice an ultrasound or x-ray is best for finding a pneumothorax but it's kind of frustrating to the rock eaters who only view it as being killed by something that is silent and invisible

**Comment by YetheSamartaka** [comment]
> > > A loud, frequent gasping sound effect for pneumothorax or making it turn the torso red/another color would be very helpful, I have lots of grug brain infantry who are having a tough time with not noticing the pneumothorax because it requires reading, and it has not been drawn in colorful crayon and is therefore invisible to them
> > 
> > 
> > Problem with pneumothorax is, that in quite a lot of cases, patient won't make any sounds, in fact when someone has pneumothorax and you will use stethoscope to listen to their lungs, doctors recoqnize it that it makes less sound than usual. I'm not an medical expert, I just tried to make some research, so feel free to correct me.
> > From gameplay point of view, yes, we are going to add some graphical indication to the medical menu, but we are also considering the function, that you would have to "check if patient has pneumothorax/hemothorax or whatever". If anything such that will be added, don't worry, it will have on/off button so each community can decide how it will work for them.
> 
> Yeah in real world practice an ultrasound or x-ray is best for finding a pneumothorax but it's kind of frustrating to the rock eaters who only view it as being killed by something that is silent and invisible

First thing would definetly be some graphical indication, similar to broken limb. Also I got idea, when speaking about broken limbs, sometimes when you get your limb broken, it will make that bone-cracking sound. Something similar could be done for pneumothorax, that the injured would make some half-breath or something, dunno, that is yet to be properly thought through.

**Comment by FluffyGhoster** [comment]
> > > > A loud, frequent gasping sound effect for pneumothorax or making it turn the torso red/another color would be very helpful, I have lots of grug brain infantry who are having a tough time with not noticing the pneumothorax because it requires reading, and it has not been drawn in colorful crayon and is therefore invisible to them
> > > 
> > > 
> > > Problem with pneumothorax is, that in quite a lot of cases, patient won't make any sounds, in fact when someone has pneumothorax and you will use stethoscope to listen to their lungs, doctors recoqnize it that it makes less sound than usual. I'm not an medical expert, I just tried to make some research, so feel free to correct me.
> > > From gameplay point of view, yes, we are going to add some graphical indication to the medical menu, but we are also considering the function, that you would have to "check if patient has pneumothorax/hemothorax or whatever". If anything such that will be added, don't worry, it will have on/off button so each community can decide how it will work for them.
> > 
> > 
> > Yeah in real world practice an ultrasound or x-ray is best for finding a pneumothorax but it's kind of frustrating to the rock eaters who only view it as being killed by something that is silent and invisible
> 
> First thing would definetly be some graphical indication, similar to broken limb. Also I got idea, when speaking about broken limbs, sometimes when you get your limb broken, it will make that bone-cracking sound. Something similar could be done for pneumothorax, that the injured would make some half-breath or something, dunno, that is yet to be properly thought through.

We could add a stethoscope (so we can reuse it later) and add the action on the chest to do the percussion if the user have it, if it's using the base settings it will just tell him there's a hemothorax, if it's using the advanced settings it will tell that it's hyporesonant (conversely in case of pneumothorax it would be hyperresonant)

**Comment by Katskan** [comment]
**Item: Burn Dressing** 
More specific to equipment that is used for burns, dry and wet burn dressings often come as large pads in sterile packaging, designed to be applied directly to a wound and secured. Real-world they are usually significantly larger than normal dressings, the ones I carry at work take the same space as about 40 regular sterile gauze pads, though I know there are smaller dressings that are on the market. Balance-wise it'd probably work to have them somewhere like 1.5x effective as the equivalent mass in regular dressings: for example if a large burn would take 3 regular dressings to stop fluid loss, then the burn dressing would have 2x mass as a normal dressing

**Comment by Katskan** [comment]
Recent discussion reminded me: often times neck wounds need occlusive dressings. Could add a chance for a "blood vessel injury" in the neck, bleeding very fast with a chance to cause cardiac arrest (they can pull in air emboli because of the venturi effect). Would be an interesting change up, and while a purpose-built chest seal won't work, a general "occlusive dressing" would work on both sucking chest wounds and vessel damage in the neck

**Comment by AVESUM** [comment]
**Description:**
Add two player stretcher carrying.

**Implementation:**
Similar to how [GGE Captive animations](https://steamcommunity.com/sharedfiles/filedetails/?id=2010224487&searchtext=GGE+captive) allow one player to "slave" the other, this could be implemented in KAT. The idea is that it is required for two players to separately interact with a stretcher;
The first player to interact gets the front carry position (kneeling with stretcher, unable to move). 
The second player is attached to the rear carry position and slaved to the first players movement along with wakling/running animation played. Either of the players can abort the carrying by "exiting the vehicle" and the movement is put to an halt in kneeling position.
Gameplay benefits would be that it is possible to walk and run with a stretcher without the immersion breaking single carrying.

**Comment by Katskan** [comment]
A Code Summary option would be very useful for training and possibly also debugging. Could simply be a view you can scroll through for past vitals checks and treatments. An example of a real one: https://i.imgur.com/g3tu9zw.jpg

**Comment by Oxan-AU** [comment]
New to Github, not sure if this is entirely appropriate, but just some comments on these suggestions from a realism perspective. Seems better to post here than the discord.

Regarding intubation. Not opposed in principle, but it's not a recommended intervention in the Tactical Field Care phase of TCCC (which I'd argue is the generally applicable context of TCCC for most Arma groups). Intubation is not even in the skill set of the majority of military medics. Extraglottic airways (in KAT, the King LT, although it should be updated to the iGel) is preferred. Or if that's insufficient, the next step would be surgical airways. So 1) I don't think intubation should replace the King LT/iGel/supraglottic airway, and 2) if a more advanced airway device than the King LT is going to be introduced, it should be surgical cricothyroidotomy.

Regarding pneumothorax. Okay, this one isn't that realistic, but just as a suggestion on ways to provide an audible cue for pneumo/haemothorax, you could consider using audio of a wheeze or crackles. Generally these might be sounds you'd hear when listening to the lungs of someone with asthma or fluid on the lungs, but sometimes they can be heard externally without a stethoscope. It's not something I've ever experienced or read about for someone with a pneumo/haemothorax, but it's one possible solution that isn't "too" farfetched.

Regarding cardiac arrest. I'd like to see less focus on cardiac arrest as a whole. The way it's represented in ACE, with chest compressions and defibrillation, is inherently flawed since it doesn't realistically reflect the reversible causes of a traumatic cardiac arrest, which is what we're talking about in the Arma context. In **traumatic** cardiac arrest, less emphasis is placed on CPR/defibrillation, with the focus being on aggressive haemorrhage control, fluid resuscitation, airway management and alleviating tension pneumothorax and cardiac tamponade. It should be needle decompression and not CPR that's getting pulses back in ACE.

----

Now I'm done complaining, I'd like to suggest:

**Description**
TXA, Tranexamic Acid. It's used to help treat/prevent haemorrhage shock by promoting the formation of blood clots to minimise bleeding, in super simple terms. The suggested implementation here is to universally extend the minimum reopen delay timer of all bandages, or dramatically reduce the reopening chance. I think it's both realistic and adds value to the simulation, and is a balanced alternative to stitching (permanent wound closure at the cost of potentially quite a long interaction v. temporarily delayed wound reopening times at the price of a relatively short interaction).

**Description**
Pneumothorax sounds. Basically what I posted above but just trying to keep the right format for suggestions. Consider using audio of a wheeze or crackles to provide an audible cue for a pneumo/haemothorax.

**Comment by Katskan** [comment]
If there's one thing I've learned working with military medical folk it's that they change radically what procedures are done by who and where based on the guidance/opinions of their medical officers. I've worked with people who as front line medics performed ETT intubation, some had to get to their Sr company medic to do anything more than NPA/OPA. Sometimes NCD is restricted to Sr medics only, sometimes medications are restricted. About the only places I've gotten info about being extremely consistent are flight medic and physician level tasks, two settings which are characterized by a maximum level of training and a maximum availability of supplies and manpower. 

I have always had the impression that KAT is extremely ambitious, in that the intent is to model as many options with as much detail as possible, and just as with real-world scenarios there are pros and cons ranging from equipment to tactical considerations to training requirements associated with each procedure that could be performed. 
For example: if you place any supraglottic airway or an ETT, that person is not breathing for themselves, which means you've now got to devote at least one soldier just to breathing for that casualty. If they have no pulse, an AED/monitor is too heavy and bulky outside of a dedicated ambulance, usually a helicopter, so now you have to dedicate a whole crew of soldiers to breathe and cycle CPR, with little or no chance of success. Just the manpower limitations are complex and severe, which is why your typical pulseless casualty is usually immediately black tagged. Even when there's plenty of resources and manpower available, trauma resuscitations often involve tactics/procedures that are pretty intense, and they could potentially be modeled by KAT but mostly are hospital kinds of stuff.

A suggestion for implementing pros/cons into game could be for example:
NPA- anyone can place, does not protect against obstruction by blood/emesis. 
IRL supply: single NPA item and lube needed.

Supraglottic- requires simple training, does protect against obstruction by blood/emesis, will cause tissue edema and necrosis over long periods (IRL ~2hr, could be shortened in-game and reflected by some kind of vitals change). 
IRL supply: single supraglottic airway item (sometimes needs 50ml syringe), tape/tube holder, must get help with breathing.

Endotracheal tube placement- requires moderate training, does protect against obstruction by blood/emesis, can remain in place "indefinitely", can be difficult or impossible in patients with massive facial trauma or burns to mouth/throat due to inhaling superheated air (something that comes to mind is burn injuries to the head or a damage threshold on the head resulting in an "airway burns" or "facial trauma" tag, which will render everything but surgical airways ineffective). 
IRL supply: single ETT, stylette/bougie, 10ml syringe, laryngoscope, tape/tube holder, stethoscope, suction, must get help with breathing.

Surgical airway- requires moderate/advanced training, does protect against obstruction by blood/emesis, can remain in place "indefinitely", is typically used to circumvent issues caused by facial trauma or upper airway burns. 
IRL supply: two philosophies, either a single device which requires cleaning the neck then inserting the device, or full kit of cleaning, scalpel, introducer, ETT, suction, tape/tube holder, may or may not need help with breathing.


A caveat to TXA, while it's awesome and realistic one of the big things with TXA is that it needs a second follow-up dose, IRL ~1 hour later, to be effective. Without the second dose it doesn't actually help anything, the only idea that comes to mind for me is when internal bleeding gets put into KAT, TXA can stop it then if it's not given a repeat within some timeframe later (customizable), the bleeding starts again, but if it's repeated it will not start again after any period

A pneumothorax shouldn't wheeze, wheezing is due to narrowing of diameter of small anatomy of the lung resulting in a wheezing sound, usually limited to exhalation, whereas on inhalation usually we hear stridor from narrowing of the glottic opening from swelling, usually in trauma this will be burns or facial/neck injuries that penetrate near the glottic opening. Crackles/rales result from fluid getting into alveoli in particular, on the interior of the lung, so are quite common in both pneumothorax and hemothorax as pressure changes and reduced circulation allow fluid to leak from blood vessels into the lung's interior, or frank blood enters the lung from penetrating wounds. Patients themselves often have both pain and feel the difficulty in breathing as these develop, and it's common to compensate and remain conscious right up to cessation of breathing entirely, followed by death without immediate management. People also often get loopy and confused or agitated as their oxygen level drops, they don't notice this but an outside observer will pick up on it.

**Comment by Oxan-AU** [comment]
We have a standard to work from in the TCCC guidelines and their corresponding skills matrix. It is the standard, so emulating that is the closest we'll get to realism in Arma and KAT. Some units do more, some less, but we can't account for every variation. TCCC is a good basis, and it's been broadly adopted globally, so it's relevant for Arma units outside of the US.

Regarding realism generally, there has to be a balance between it and fun. Realistically, anyone getting a tourniquet, or IV fluids, or morphine, or any degree of airway management, is going to be evacuated. But many Arma units won't have the capacity to be evacuated every single wounded player, and who wants to spend all the time getting evacuated away from the fun and action just because realism? So regarding TXA as an example, yeah, realistically, you need a follow up dose. But from a gameplay perspective, does that really matter? Adding single dose TXA (as an example) might add a useful intervention, and knowing when to use it adds an extra challenge for the Arma medic - who is almost always a layperson. They shouldn't have to be an actual healthcare professional to play medic in Arma.

Ultimately, what I find most appealing about KAT and the reason why I think it's a great mod is it just adds much needed variety to ACE medical. It's challenging because you have more options. It will never be 100% accurate, nor should it. It should be based in reality but it needs to be fun. I don't find the idea of clicking my mouse every five seconds to deliver IPPV fun.

Regarding lung sounds, yeah it's not entirely accurate like I said, but it is a reasonable audible cue for someone playing a medic who's not actually trained IRL. Wheezing is something most people are able to recognise, and I'd suggest most people would associated it with a lung problem, so it's reasonable from a gameplay perspective.

Anyway, I can't code, so all of this is just my humble suggestion for those with the talent to potentially implement it.

**Comment by mazinskihenry** [comment]
This was discussed a bit on the discord, but I wanted to add it here for the record. 

**Naxolone**
One of the issues in ACE Medical is the power of morphine and the threat of overdose. Currently, the only way to handle overdoses is to feed a patient epinephrine every two minutes until the morphine clears through (which will usually take 10-15 minutes) feels disempowering to both the medic and the patient. It leaves both to sit on the ground and stay out of the fight with nothing to do but once again give them epinephrine to prevent them from passing out. 

It would be nice to have naxolone to counter this type of situation, where a patient has been given one too many morphine and a medic can come by and "clear" the morphine from the patient. From my understanding of the ACE medical system, this would be as easy as going into the list of medications that a patient has and deleting the morphine from that list to prevent it from having any more effect on them. Or it could be a timer or sorts, where it directly counters the morphine, but only for a limited amount of time as to allow for some sort of punishment when someone has an excessive overdose. It just would be nice to have that sort of direct counter to morphine that is (sort of) realistic and allows for medics to correct for mistakes during treatment to make sure that patients don't stay on the ground too long and to have another tool in their kit that they can use.

**Comment by Katskan** [comment]
If we get new medications, it would be nice to have a system where meds can be taken in vials which allow multiple uses and more customized dosing, while having a smaller package than auto-injectors. For example in the real world I often give 4-5 doses (to the same patient) out of a morphine or fentanyl vial, which combined with the syringes needed is about the same size as a single auto-injector would be. Vials could be a nice system which rewards having better training/knowledge by reducing the size and weight of medications in your kit, especially if there's more medications needing to be stocked

**Comment by Katskan** [comment]
**_Injury Types:_**
**Neck Injury**
Injuries to the neck which penetrate the platysma muscle have a high frequency of complication by severe blood loss and air embolism due to vascular injury, as well as pneumothorax from infiltration of air into the chest cavity via spaces connecting to the neck. 
-Chance per second to cause tension pneumothorax
-Chance per second to cause sudden cardiac arrest

**Evisceration/Open Abdominal Injury**
Abdominal injuries that penetrate fully through the muscles of the abdominal wall can result in either exposure or herniation of organs.
-Extreme pain for patient
-Over time, reduces BP
-Immediate maximum fatigue/forced walk

These injuries, while severe and in real-world scenarios would necessitate rapid surgical intervention, are similar in that way to injuries like hemothorax and traumatic cardiac/respiratory arrest, which would also be catastrophic for a casualty. They could also add some flavor to otherwise mundane injuries to the head/chest, and could be very manageable with supplies by changing to more niche chest seal to a general occlusive dressing, which would be appropriate for thoracic, abdominal and neck injuries.

**Comment by Oxan-AU** [comment]
This was raised briefly in the Discord but would like to have it recorded here as well.

**Fresh whole blood and walking blood banks**
TCCC guidelines recommend the administration of fresh whole blood, taken from pre-screened donors in the field (when cold stored blood is unavailable). Having the ability to draw blood from a healthy player to then be able to transfuse into another would add another level of realism and complexity in terms of both clinical management and resource management.

Current real world guidelines state around 450 ml of blood would be taken from the donor who can then return to the fight.

One question that stands out is how would this ability be balanced? Does the donor actually lose their blood as far as ACE is concerned? Do they lose only half what is actually taken, to balance them being "healthy" and uninjured? Do they not suffer any negative affect at all?

**Comment by JustinD39** [comment]
As Oxan-AU previously suggested TXA 

"Description
TXA, Tranexamic Acid. It's used to help treat/prevent hemorrhage shock by promoting the formation of blood clots to minimize bleeding, in super simple terms. The suggested implementation here is to universally extend the minimum reopen delay timer of all bandages, or dramatically reduce the reopening chance. I think it's both realistic and adds value to the simulation, and is a balanced alternative to stitching (permanent wound closure at the cost of potentially quite a long interaction v. temporarily delayed wound reopening times at the price of a relatively short interaction)."

I believe a potential way to implement this could be by administering the drug could A) lower the reopen chance of unstitched wounds B) slow the bleed out rate for open wounds. C) possibility for new minor wounds to automatically "bandage" to simulate increased clotting in the blood.  A drawback to administering TXA could be the potential for hypotension since administering the medication above 50milligrams a minute can cause hypotension.

**Comment by Katskan** [comment]
An icon that shows up on the body image display in the medical menu for pneumothorax/tension/hemopneumo. Could appear either on the chest of the patient or in one of the corners of the box, if on top of the patient chest could use something like a purple, green or brown icon which would stand out against any of the regular color backgrounds, perhaps all 3 to indicate different conditions, i.e. green is regular pneumo, purple is tension, brown is hemopneumo. May also use a similar format for airway issues displaying an icon on/near the neck. Could potentially try and do a system where these conditions are hidden until the examination actions are performed to check airway, then the icons show up on the medical menu's display.

**Comment by Oxan-AU** [comment]
From the Discord...

Just regarding recognising dead-dead in an immersive way, would it be possible to add an extra fuction to the AED-X (+/- normal AED, most professional models have rhythm monitoring capability anyway) to "Check Rhythm". Very simplistic but "Asystole" ("Flatline") could be the result if the casualty was dead-dead, and "Pulseless Electrical Activity (PEA)" could be the result if the casualty was in cardiac arrest, but not dead-dead.

- I say PEA because that's usually the initial presenting rhythm in a traumatic cardiac arrest, i.e. battlefield trauma
- I recognise PEA's are not shockable rhythms, so defibrillation with the PEA doesn't make sense. So I suppose a "Non-Shockable" i.e. dead-dead and "Shockable" i.e. reversible cardiac arrest, check could also be an option.

Again, a more immersive way of recognising death as opposed to judging ragdoll positions or testing if you can drag/carry...

**Comment by MAJThomasASOT** [comment]
**Description:**
- In regards to blood type, clients can abuse the system by being able to change their blood type half way through a mission. So let's say medics are out of a particular blood type, then the person requiring treatment can simply go into the addon options, change their blood type and then be treated. 

I feel like this is a flaw in the server/client settings. To fix this, I believe there should be an option to force the blood type setting on the server that overwrites the client setting. You could have an option in the server blood type setting that is 'Random'. Then have the option to overwrite client settings. 

With this random option, mission makers can then choose to allow clients to change their blood type, whilst others can choose to not have this exploit a possibility.

- Forcing one blood type literally removes the need for the other ones. So therefore a forced random option should be available. 

- I acknowledge the fact you should be able to trust people to do the right thing, but unless you can guarantee they will with a setting, there's always that concern of those who will do the "wrong thing".

**Comment by MAJThomasASOT** [comment]
Probably going to stir the pot a bit here but after reading a majority of these suggestions I just wanted to mention that there needs to be a line drawn when trying to simulate some of these features and implement them in game. Some of these suggestions are going to require extensive amount of work and documentation to teach people how they work. (Don't get me wrong I love this stuff) but you have to consider was is feasible and reasonable in game.

Just to state my experience, currently serving, combat first aider (a basic medic), armoured crewman. So I understand the suggestions being made medically. 

Just as a few examples off the top of my head;

- AED only shocks automatically: I get that IRL it does this, but removing the option for the treater to choose simply removes an action they get to do. Yeah, it's only clicking a button. But it's a step they get to involve themselves in rather than it being automatic. So although not realistic, some level of realism sometimes need to be sacrificed for the purpose of game play. (The idea of a manual AED and an 'actual' AED might be a good compromise. So the higher level medical personnel can use the advanced/automatic one.)

- Pneumothorax: Another topic that has caused a lot of discussion. Yes, it's a complicated injury to ID, treat correctly in the field etc etc. But I feel adding too much to the process of treatment is going not only cause the treatment time to be heavily extended, but also cause further performance issues. Again, don't get me wrong, would be great to have all treatment and symptom options. But due to some of the suggestions, the realistic ability of ARMA and the work load on the KAT dev team needs to be considered more. 

A few ideas/mods/extensions have been made already which I personally feel have targeted some of these suggestions at an appropriate game play vs realism level. 

Blood donor and blood types: [https://steamcommunity.com/sharedfiles/filedetails/?id=2441179531  ](url)
Pneumothorax and extended airways: [https://steamcommunity.com/sharedfiles/filedetails/?id=2448877245](url)

I mean no disrespect to anyone who makes suggestions or comments and some of them are brilliant ideas. I just see it too often when people get carried away with requests and don't think about the feasibility and balance of game play and realism. Especially IRL medical personnel who are excited to implement all forms of assessment and treatments. 

In my unit we aim for a down time of a patient of 5-10mins. Any longer and it starts to become a severe pain for those members going down. If you continue to add numerous features for longer/more in-depth treatment, considerations need to be made regarding settings to enable adjustment to allow the desired overall treatment/down time for some units. 

Wow, this turned out way longer than I anticipated. I'll finish up here. Again, I don't mean any disrespect and I think the suggestions should keep coming. I just think some features should be strongly considered and prioritised so KAT doesn't become a performance monster and/or we burn out the KAT staff. Having a medium complex, working mod is better than a broken super complex mod in my opinion.

**Comment by Will-Nichols** [comment]
I'm going to clip a few points here and touch on some of what @MAJThomasASOT mentioned above.

> 
> 
> Some of these suggestions are going to require extensive amount of work and documentation to teach people how they work. (Don't get me wrong I love this stuff) but you have to consider was is feasible and reasonable in game.
> 
I love this statement. Teaching other players is a huge aspect of the Arma 3 community. However it does get ridiculous when trying to learn and teach everyone all the time. At which point you get to a place where you have zero time to play yourself and you get burned out. Reasonable amounts of learning a new skill and somewhat dumbed down roles make sense in a game. 

> 
> In my unit we aim for a down time of a patient of 5-10mins. Any longer and it starts to become a severe pain for those members going down. If you continue to add numerous features for longer/more in-depth treatment, considerations need to be made regarding settings to enable adjustment to allow the desired overall treatment/down time for some units.
> 
>

I agree but I love it when you have to actually transport patients who are "too screwed up" to fix in the field. It adds a level of realism that I like. Especially on larger maps where you need to transport some caualty 15-20 minutes by helo just to get to a field hospital. Those are the best parts of dealing with medical in a game for me. That also allows the dudes that didn't get hurt that bad to try and figure out how to continue the mission while missing a few folks for a while.

Overall you bring up some great points and I am glad to read them.

**Comment by Will-Nichols** [comment]
> 
> 
> From the Discord...
> 
> Just regarding recognising dead-dead in an immersive way, would it be possible to add an extra fuction to the AED-X (+/- normal AED, most professional models have rhythm monitoring capability anyway) to "Check Rhythm". Very simplistic but "Asystole" ("Flatline") could be the result if the casualty was dead-dead, and "Pulseless Electrical Activity (PEA)" could be the result if the casualty was in cardiac arrest, but not dead-dead.
> 
>     * I say PEA because that's usually the initial presenting rhythm in a traumatic cardiac arrest, i.e. battlefield trauma
> 
>     * I recognise PEA's are not shockable rhythms, so defibrillation with the PEA doesn't make sense. So I suppose a "Non-Shockable" i.e. dead-dead and "Shockable" i.e. reversible cardiac arrest, check could also be an option.
> 
> 
> Again, a more immersive way of recognising death as opposed to judging ragdoll positions or testing if you can drag/carry...

If you can't judge someone as dead via the ACE response system and using TCCC triage system with the various colors...why do you need to hook up an AED to see what you are really needing to do based on a judgement by the medic on scene? If you have a dude who stepped on a landmine the best thing do is to move on; similar effect to someone that took a .50 round to their head/chest in the game. IF you can get a pulse and they are breathing; regardless of level of consciousness in game then you should be able to stabilize at least and prepare for transport to the med/aid station in the rear. Otherwise don't waste time and resources by watching your character apply/setup the AED just to tell you they are dead.

**Comment by Oxan-AU** [comment]
Most suggestions are mostly adjuncts to treatment that simply provide alternative options or enhance existing ones. Few really enforce any extra steps that must be taken to achieve the same outcome. The Check Rhythm suggestion was presented as an additional option to recognise cardiac arrest from dead-dead, not as a prerequisite step before using the defibrillation action. It's a semi-regular occurrence to have someone drop by the Discord either directly asking how to recognise dead-dead, or otherwise demonstrating they're not quite sure.

Also regarding Battlekeeper's extensions - every feature request in this thread was made prior to Battlekeeper producing his work. Many requests are now redundant, since he's introduced those features.

**Comment by Katskan** [comment]
An option for using items found in a vehicle inventory before using items in patient/provider's inventory while mounted would be awesome. Allows for extremely light dismounted kits without fiddling with jump bags or transferring items to/from vehicles during treatment

**Comment by Paramed101** [comment]
### Visible triage card on player

**Description:**
Create a visible tag attached to a player with the colour corresponding to the Triage Card of the player it is attached to.

This is visible as an object attached to the player. normally placed around the neck (preferably) or a limb by an elastic band/rope. Changing the triage card on the medical menu will replace the visible triage card. If "none" is selected from the triage menu, then no item is attached to the player.

Refrence image [https://www.rcemlearning.co.uk/wp-content/uploads/image-2.png](https://www.rcemlearning.co.uk/wp-content/uploads/image-2.png)


**Benefits**
Benefits include being able to look at players and seeing the already assigned Triage states. This allows the passing of care to be faster and medics know who is at what level when dealing with post-triage Mass-cas.


**How to implement:**
Check the state of the player's triage card and attach items based on state. This script can be run each time a players triage card is updated.

I have little to no experience coding but I'm guessing that there is a way to do this as there is an attached item module. I'm assuming the script will be similar and call similar functions.

**Comment by MAJThomasASOT** [comment]
> ### Visible triage card on player
> **Description:**
> Create a visible tag attached to a player with the colour corresponding to the Triage Card of the player it is attached to.
> 
> This is visible as an object attached to the player. normally placed around the neck (preferably) or a limb by an elastic band/rope. Changing the triage card on the medical menu will replace the visible triage card. If "none" is selected from the triage menu, then no item is attached to the player.
> 
> Refrence image https://www.rcemlearning.co.uk/wp-content/uploads/image-2.png
> 
> **Benefits**
> Benefits include being able to look at players and seeing the already assigned Triage states. This allows the passing of care to be faster and medics know who is at what level when dealing with post-triage Mass-cas.
> 
> **How to implement:**
> Check the state of the player's triage card and attach items based on state. This script can be run each time a players triage card is updated.
> 
> I have little to no experience coding but I'm guessing that there is a way to do this as there is an attached item module. I'm assuming the script will be similar and call similar functions.

https://steamcommunity.com/sharedfiles/filedetails/?id=2053966331&searchtext= 

That might cover what you're after mate. We use it and it's great.

**Comment by Paramed101** [comment]
> > ### Visible triage card on player
> > **Description:**
> > Create a visible tag attached to a player with the colour corresponding to the Triage Card of the player it is attached to.
> > This is visible as an object attached to the player. normally placed around the neck (preferably) or a limb by an elastic band/rope. Changing the triage card on the medical menu will replace the visible triage card. If "none" is selected from the triage menu, then no item is attached to the player.
> > Refrence image https://www.rcemlearning.co.uk/wp-content/uploads/image-2.png
> > **Benefits**
> > Benefits include being able to look at players and seeing the already assigned Triage states. This allows the passing of care to be faster and medics know who is at what level when dealing with post-triage Mass-cas.
> > **How to implement:**
> > Check the state of the player's triage card and attach items based on state. This script can be run each time a players triage card is updated.
> > I have little to no experience coding but I'm guessing that there is a way to do this as there is an attached item module. I'm assuming the script will be similar and call similar functions.
> 
> https://steamcommunity.com/sharedfiles/filedetails/?id=2053966331&searchtext=
> 
> That might cover what you're after mate. We use it and it's great.

Yeah, it will do the job, I'm gonna try it out. Thank you.

Would still be nice to have it as an object tho, instead of just on screen.

**Comment by MAJThomasASOT** [comment]
Object might be a bit hard to see in my opinion. Plus doing an object would most likely mean an attachTo command which isn't ideal for server performance. Especially at the size we run at. Haha

**Comment by Katskan** [comment]
Very much in need of either a setting for a lower limit to O2 without making it a lethal O2 value, or an item/setting that will increase rate of O2 gain. It's super rough to not have the lethal O2 value set, then getting 1% O2 per 10 seconds.

**Comment by Katskan** [comment]
Would it be possible to make the stretchers as items that can be attached to a vehicle, the way you can do with things like chemlights and strobes which can be attached and anchored in place on a vehicle? It could help with some utility in directly loading the stretchers into vehicles, that way the stretcher doesn't slide off with g-force or desync.

**Comment by FPFalcon** [comment]
New Icon for KAT discord and KAT mod

Developed through feedback on the discord, I thought the icon for KAT and by extension the one in ArmA, needed updating since KAT is more than just adding a KingLT. So I took it upon myself to develop a new one. Use it, don't, it's up to you.

[KAT ICON FINAL.zip](https://github.com/Tomcat-SG/KAM/files/8544729/KAT.ICON.FINAL.zip)

**Comment by fijik124** [comment]
IO and IV system

Hi I have 1 big problem with the new IO and IV system IO in this mode means fast IO it is a tool that you "shoot" into the sternum transfusion with a hook that catches the bone and has no chance to fall out but IV is a classic transfusion needle that is inserted into a vein so at the slightest movement or pulling it can loosen or fall out 

My question is whether it would be possible to divide this setting into IV and IO in the future because it is quite unrealistic and at the same time quite annoying when it keeps falling out I understand it's some kind of gimmick but we try to stick to reality so I wanted to ask if it's in the plane somehow split this setting

Thank you for your reply and have a nice day

**Comment by yesthatRiVin** [comment]
Add an option to check for cyanosis. Have it so medics & bystanders can get a rough estimate of someone's blood o2 content without a pulse oximeter. Base it off of percentages of the lethal 02 variable, that way it can change depending on what the server is set to.

It would just be a 2-second interaction on the head and would return one of four values:
- "No cyanosis": 100% o2 value
- "Slightly cyanotic": 99% to 166% of the lethal o2 variable
- "Mildly cyanotic": 165.99% of the lethal o2 var to 133% of the lethal o2 var.
- "Severely cyanotic": 132.9% of the lethal o2 var to 100.99% of the lethal o2 var.

If you have any questions I'm on the KAT discord as RiVin.zip

**Comment by yesthatRiVin** [comment]
At the suggestion of one of my unit’s medics:

-Change the name of “turn head” to “Recovery Position”

-Change the AAT kit to two devices, the NCD Kit and the Chest Tube
     NCD Kit works similarly to the AAT kit with the exception of it doesn’t have a drain fluid option. It will also have a slight chance to have the tension come back.
     Chest Tube takes 30s to put in but definitive clears both tension and hemo ptx

**Comment by Nova0127** [comment]
Add an option to enable/disable permissions (Medic/Doctor, Doctor only, Anyone) for injection of some of the new merged drugs. Similar setting to AED/AED-X. Probably only Ammonium Carbonate, Naloxone, and TXA? Maybe could extend to Epinephrine and Morphine as well.

My unit allows everyone to use Epi and Morphine for quality of life, but we think the other more advanced drugs should be able to be restricted as they probably would be IRL. Also seeing as they only really enhance a medic’s ability and aren’t essential for stabilization (Non-medics would still be able to provide basic care).

**Comment by yesthatRiVin** [comment]
_It hasn't been 24 hours yet polski ;)_

Add ketamine as an IV med

Recomended values:

painReduce = 1.7
hrIncreaseLow[] = {5, 10};
hrIncreaseNormal[] = {10, 15};
hrIncreaseHigh[] = {15, 20};
timeInSystem = 600;
timeTillMaxEffect = 15;
maxDose = 4;
incompatibleMedication[] = {};
viscosityChange = 10;

And for some reason, I couldn't find a BP increase variable but it should increase RR by 15%

**Comment by yesthatRiVin** [comment]
Add a "pray to the dark gods mechanic"

It works in two parts, an inventory item known as the "sacrifice kit", and a launcher item named "sacrificial goat".

In order for it to work, you have to first use the ace self-interact menu while the sacrifice kit is in your inventory to place the sacrifice kit on the ground. It then will place down the 3d model of a terribly drawn pentagram with candles on every point in the star. 
Then, you press the ace interact key on the pentagram while you have the goat in your inventory and select "prepare sacrifice". It will then remove the goat from your inventory and place the model down in the middle of the pentagram. (you can not place the goat down until you have placed the kit, think like how CSWs work, where you have to place the base and then the weapon)
The player will then open the ace interact menu on the pentagram again and see the option to "pray to the old ones". If they do it, most of the time the goat and pentagram will despawn and everyone within 5m of the pentagram is fully healed, regardless of their status.
However, there is a 1 in 20 chance that instead of healing, lightning will strike the pentagram and everyone within 5m will die instantly, damned for all eternity for meddling with forces beyond their comprehension.

I'm only suggesting this half as a joke and half as a decent idea for a fun Halloween/April Fools gimmick.

**Comment by yesthatRiVin** [comment]
Add syringes. 

ACE already has the code to combine items through the ace interact menu (chemlights and chemlight shields) so it *should* be possible to create a system where you have an item "syringe" that you can combine with a drug to create a "<drug> syringe". It would work under the same conditions as an IV, i.e. it needs a heartbeat to work, and the arm can't be too damaged. Once it's prepped, it should show up in the ace medical menu in Advanced Treatment, the same as IV. It'd obviously be disposable. That way medics can have a way to administer "unfucking shit" drugs like TXA without having to establish an IV line beforehand, allowing quicker treatment time for critical patients. 

Alternatively, if that's a bitch to code you could just create syringes that are prefilled with critical drugs like TXA, Amio, Ketamine (once you finally add it), etc. That way you could code them basically as an auto-injector that only works when the limb satisfies the "IV works" condition. Some medics might actually prefer that because it means they don't have to go back to the ACE self-interact menu if they didn't prep enough syringes before treating.

**Comment by Ghostworrior** [comment]
Add drop rates/pressure infuser.

It would be cool specially for an feature already present in ACE Pharm IV clotting, to get the option to change Infusion drop rates.

Normal Infusions run like they do now if fully opened, with only half the flow rate if half closed and so on.
I would suggest values of Running free, half closed, slightly dripping.

On the other hand I would like to see the implementation of pressure infusers, which would allow the medic to set a pressure as he desires.
The value will be check against the current RR of the patient. ((sys value / sys rr)+(dia value / dia rr))/2 the result would be applied as a multipler for the flow rate.
All chances and other variables are examples and would be awesome if editable in the addon options.

**Comment by Ghostworrior** [comment]
Change AAT Kit and add NCDs/drainage sets.

I would suggest that we separate the NCD needles and the drainage sets, which currently are within the AAT Kit.
This would allow medics and community's more freedom: i.e. having a frontline medic being able to fix a tension pneumothorax but unable to fix a hemopneumothorax, which would require higher medical aid...

The before mentioned NCD needles could come in to types:
1. proper NCD device with an 95% success chance.
2. Improvised NCD i.e. 14G IV needle 50% success chance.

The drainage set would be a straight forward replacement for the AAT Kit with only one function.

For the moment I would remove the AAT Kit till my next idea would be in the ring.
All chances and other variables are examples and would be awesome if editable in the addon options.

**Comment by MiszczuZPolski** [comment]
> We already have IV/IO flow rates dependent on circulation... Now I would be cool specially for an feature already present in ACE Pharm IV clotting, to get the option to change Infusion drop rates.

IIRC there isn't flow rate dependent on circulation in KAT Pharmacy (except for one addition - if pulse <20, flowrate/1.5)

About rest of request - i would like to see some more explanation/description how things should work - you can ping me on discord if you want - easier to discuss things there

**Comment by Ghostworrior** [comment]
Airway Management Extension.

The next big thing you devs are going to hate me for...

Escalation Airway Management:
Till the KingLT all stays the same.
The King LT gets a 10% chance to fail placement, has the option to reapply without consuming the KingLT.

There is a 5% chance of the patient getting an difficult airway parameter upon going unconscious.

A patient with difficult airways can either be ventilated with a KingLT but with half capacity only (SpO2 change gets halfed) 
or cant be ventilated at all, chance for both could be standard 50/50.

If the patient cant be ventilated you can get an AAT or Surgical Airway Kit (naming change could be good to avoid confusion)...
which acts as the KingLT does now, completely ignores/removes occlusion and obstruction, but induces slight pain. 

All chances and other variables are examples and would be awesome if editable in the addon options.
Thanks for your work guys... If I would be more competent I would try to help but...

**Comment by Paramed101** [comment]
**Description:**
Feature request for AEDs to provide feedback.

The use of an AED is to provide Automated check on the patient's heart activity and suggest if a shock is practical.
Every AED would do this and provide audio feedback like this:
https://youtube.com/clip/Ugkx0j6qGgcTWG9dxJ5I_suPX0CzxEEeCwAU

HOWEVER, AEDs generally do not provide SPO2 or pulse feedback. This would be the benefit of an AED-X.

**Currently**
Currently, the "normal" AED doesn't provide any assessment or audio feedback.
The AED-X is able to provide audio feedback and tell you if it's a shockable rythem.

**Feature request**
Add the ability for the "normal" AED to "check rhythm". Do NOT add the ability to "monitor vitals" as this would be reserved for the AED-X.

**Reason**
Without this feature, it is required that you use an AED-X if wanting to use the advanced cardiac system, making the AED almost pointless.
By adding the feature, it actually makes the AED a useful alternative to the AED-X, but still make the AED-X the best option.
It balances the equipment and makes it work like real life.

**I have seen this has already been requested** Im requesting again ato show interest

**Comment by yesthatRiVin** [comment]
Credits to ArcherAdaptive for this idea.

Have blood expire after a certain amount of time. It's unrealistic to have units out there running around for hours with blood in their packs with no downsides. So have it so unless blood is stored in a medical vehicle, it expires (removes itself from the users inventory with a popup saying "<#>mL Blood has expired"). The default value should be 600 seconds, with the option to change that in settings.

**Comment by Will-Nichols** [comment]
> Credits to ArcherAdaptive for this idea.
> 
> Have blood expire after a certain amount of time. It's unrealistic to have units out there running around for hours with blood in their packs with no downsides. So have it so unless blood is stored in a medical vehicle, it expires (removes itself from the users inventory with a popup saying "<#>mL Blood has expired"). The default value should be 600 seconds, with the option to change that in settings.

The rule for real world is not having RBC units out of temperature controlled storage is 30 minutes and not longer than 4 hours for a complete transfusion of the RBC unit. In my humble opinion the default should at least be 30 minutes (1800 seconds) and have it as a changeable option in the settings because not all units use the "blood" within the 10 minute mark and several don't use it at all because they use the "plasma" and "saline" from ACE because it acts the same as "blood".

**Comment by Katskan** [comment]
Perfusion system overhauls/expansions

Part 1:
Set CPR heart rate to 100-120

Part 2:
Raise minimum heart rate to lose SpO2 from poor perfusion to 50. Replace the current system where O2 loss occurs when BPM is under 20, to any pulse where BPM is under 50 O2 is lost.

Part 3:
Add a check that if the patient is conscious, if their BP is <90 systolic then they lose SpO2. Additionally, allow being made conscious by carbonate or slaps based on vitals only, not blood volume. 

Why? This does 2 things that bring positive depth to the system. First, it makes medications that primarily alter BP have function beyond manipulating IV flow rate. Second, it gives experienced medics options to use tools to wake patients early, provided they can manage the resultant consequence in terms of burning O2.

Also who does CPR at 30bpm. Arma characters need to have a mass re-cert in BLS.

**Comment by Katskan** [comment]
**Organ damage types as a surgery expansion:**
Add organ damage types which can occur similar to how pneumothorax is done, where depending on the part of body hit, above a certain damage threshold you roll a chance for different organ issues.
Example ideas:
Chest:
Heart- causes severe pain, rapid SpO2 loss, continuous rapid blood loss until treated
Lung- current pneumothorax system, add continuous slow blood loss until hemothorax is treated
Liver- rapid blood loss until treated, slow increase to acidosis level
Kidney- rapid increase to acidosis level

Head:
Major vessel injury- rapid blood loss until treated
Brain damage- high chance of losing consciousness until treated, increased vomit chance when unconscious

**How to treat:**
Add new tools:
1- vessel repair kit
2- wound suction kit
3- graft mesh
Example procedures:
Heart repair- Incision, retract skin, cut bone, retract bone, suction blood, vessel repair, graft mesh, bone plate
Liver repair- Incision, retract skin, suction blood, vessel repair, graft mesh
Kidney repair- Incision, retract skin, vessel repair
Major vessel injury- Incision, retract skin, vessel repair
Brain damage- Incision, retract skin, cut bone, retract bone, suction blood, vessel repair, bone plate

How to balance:
With configurable damage thresholds and occurrence chances, these can be eliminated from gameplay or be made relevant as consequences for units which do not use instant death settings. Additionally, for units which have rear-line support units dedicated to in-depth medical performance, it allows for medical evacuation to serve a purpose beyond simply RP or evacuating to a designated area where the newer bone and debridement options can be performed. 
Disclaimer: heavily inspired by Space Station 13's organ and surgical system.

**Comment by Katskan** [comment]
Deteriorating fractures. 

Like it says on the tin. 

An option to have simple fractures be made worse by moving around too much while they're not splinted.

**Comment by Katskan** [comment]
Add a setting to allow time penalties/boosts on actions like ptx/tptx/htptx/surgery for lower medical level. Basically setting so "doctor" tagged personnel can do stuff faster than "medic".

**Comment by Th3Jock3R** [comment]
Hallo,

first thank you for continue the development of this great mod... We realy appreciate it!

Second, to my request:
It would be very nice if there is a setting option in the CBA settings for the KAT blood groups, as what the normal ACE blood items without blood group are treated.

My suggestion for this would be the setting options:

0: that means it is compatible with all Bloodtypes
random: That means when creating the transfusion, it will be decided internally which blood type it is. (good for survival missions, where unidentified blood can be found)
Incidentally with Incompatible: Here, there would be a risk for AB (& AB+) to get an incompatible blood unit because either donors blood unit is a rare blood group which cannot be mapped within the AB rhesus factor system or the unit is simply spoiled.
Incompatible: are always incompatible, RP technically corrupted or contaminated in end time scenarios like Exile or similar.
Thank you for you great work and wish you all a nice Christmas time!

**Comment by MiszczuZPolski** [comment]
So basically, you want new items like "Unidentified blood" and "Spoiled blood"?
ping me on kat dc, its easier to discuss

**Comment by Th3Jock3R** [comment]
~~link?~~

already found it... Sry, Layer 8.....

**Comment by MiszczuZPolski** [comment]
Suggestion made by [NMoreira88](https://github.com/NMoreira88) in #77 

1. Add the option to raise victim's legs.
a) If the victim has low blood volume but is not able to receive fluids due to inexperience, lack of permissions or no equipment available, there should be the option to raise his legs in order to concentrate fluids in his vitals.
b) If the victim has increased heart rate due to low blood pressure this would also help slowing it down, specially important if the victim has wounds that aren't closed or torniqueted limbs without bandaged wounds.

2. Adding a breathing feature to victims, to better complete the SpO2 mechanic.
a) Right now there's nothing related to breathing except for pneumothorax injuries and airway management. Implementing a breathing function would make triage a lot more decisive. According to NAEMT TC3 ASM victims that are unconscious and not breathing shouldn't be prioritized over other types of casualties that retain the power to breathe, dispite responsiveness and consciousness. This is obviously when under fire and not in an ER or Battalion level Trauma unit and normally a guideline for Combat Lifesavers.

**Comment by MildlyInterested** [comment]
If person has a broken or splinted leg their leg apparently is (temporarly) shorter than the other one. 

Since their stride is now shorter on one side the patient should start walking slightly in circles and can't run straight anymore without A/D input.
That's my suggestion.

**Comment by GoldenTomcat** [comment]
1) Add a scrollbar to the medications tab. The medications tab allows for a maximum of 11 items. This limits what you can carry and use if you need to carry 12 or more medications.  

2) Add an ability to differentiate which arm the pulse oximeter is on. At the moment, the "remove pulse-oximeter" option displays on each arm irrespective of where the psychical device is. This is a problem in a scenario where you have tourniquets on each arm and you want to remove the one blocking the pulse oximeter. This ends up in you having to guess which arm to remove the tourniquet from. 

Adding a line in the overview, similar to how the tourniquet or IV/IO needle shows up, would be an ideal solution.

**Comment by MiszczuZPolski** [comment]
1) is not easy thing, that require messing with arma GUI - and nobody really want to do it
2) will be added "soon" by Blue

**Comment by GoldenTomcat** [comment]
Yeah, I had a feeling that was going to be off limits, no bother though. 

Good news on the second one though, thanks!

**Comment by Anemic1** [comment]
Here are a couple suggestions.

**Add more in-depth treatment to burns.**

 -Burns should cause tachycardia and hypotension secondary to hypovolemia.  This is typically treated via fluid resuscitation using the parkland formula. Burn sheets would be a nice addition too.
 
 -Burns that affect the airway can rapidly induce swelling and cause closing of the airway if an advanced airway adjunct is not in place.  
 
 **Airway**
 
 -For low skilled providers BVM usage can be over zealous with ventilations, and without an advanced airway adjunct in place this could cause air to enter the stomach and cause gastric distention. This should cause emesis to enter the upper airway more frequently.
 
 -It says that intubation is added with this mod, however, I do not see endotracheal intubation in-game, only supraglottic airways. ET intubation allows for a vastly more secure airway and prevents vomitus and blood from entering the lower airway. 
 
 -surgical cricothyrotomy should be an option for severe trauma or airway obstruction. 
 
 **hemorrhaging**
 
 -unstable closed fractures should cause internal hemorrhaging.
 
 -Add the application of manual traction or using a traction splint for midshaft femur fractures. 
 
 -Add a pelvic binder for an unstable pelvis fracture.
 
 -Tourniquets that are applied for extended periods of time should cause rhabdomyolysis when removed.
 
 
 **Misc**
 
 -add sodium bicarbonate for treatment of metabolic acidosis.  
 
 -IVs should not be allowed in a limb with a fracture as it could cause extravasation. 
 
 -Please add a notification when people enter your medical menu. _I've been drugged to many times against my will ;'(_

**Comment by Will-Nichols** [comment]
> Here are a couple suggestions.
> 
> **Add more in-depth treatment to burns.**
> 
> -Burns should cause tachycardia and hypotension secondary to hypovolemia. This is typically treated via fluid resuscitation using the parkland formula. Burn sheets would be a nice addition too.

**How about COULD CAUSE instead of SHOULD CAUSE; It's not always a if this then that response by the human body to a trauma event.**

> 
> -Burns that affect the airway can rapidly induce swelling and cause closing of the airway if an advanced airway adjunct is not in place.
> 
> **Airway**
> 
> -For low skilled providers BVM usage can be over zealous with ventilations, and without an advanced airway adjunct in place this could cause air to enter the stomach and cause gastric distention. This should cause emesis to enter the upper airway more frequently.
> 
> -It says that intubation is added with this mod, however, I do not see endotracheal intubation in-game, only supraglottic airways. ET intubation allows for a vastly more secure airway and prevents vomitus and blood from entering the lower airway.
> 
> -surgical cricothyrotomy should be an option for severe trauma or airway obstruction.
> 
> **hemorrhaging**
> 
> -unstable closed fractures should cause internal hemorrhaging.

**Same thing as what you say about burns; unstable closed fractures `CAN POSSIBLY/COULD CAUSE` instead of SHOULD CAUSE; It's not always a if this then that response by the human body to a trauma event.**

> 
> -Add the application of manual traction or using a traction splint for midshaft femur fractures.
> 
> -Add a pelvic binder for an unstable pelvis fracture.
> 

Limitations of the Arma 3 engine don't allow for details like femur fractures of pelvic injuries. It's just not possible.

> -Tourniquets that are applied for extended periods of time should cause rhabdomyolysis when removed.
> 
> **Misc**
> 
> -add sodium bicarbonate for treatment of metabolic acidosis.
> 
> -IVs should not be allowed in a limb with a fracture as it could cause extravasation.
> 

Again limitations of the Arma 3 engine on the IV placement and bicarb is already in one of the drug sections of the mod or a supporting mod.


> -Please add a notification when people enter your medical menu. _I've been drugged to many times against my will ;'(_

**Comment by Anemic1** [comment]
> How about COULD CAUSE instead of SHOULD CAUSE; It's not always a if this then that response by the human body to a trauma event.
> 

You are right, my verbiage was wrong.

>Limitations of the Arma 3 engine don't allow for details like femur fractures of pelvic injuries. It's just not possible.
>

Good to know, I'm very ignorant when it comes to coding. 

>Again limitations of the Arma 3 engine on the IV placement and bicarb is already in one of the drug sections of the mod or a supporting mod.
>

I might be blind but I cannot find bicarb anywhere. I have also looked around the discord and on the workshop for supporting mods that might contain it.

**Comment by garytott** [comment]
EXPAND COAGULATION USER SETTINGS - allowing the user to adjust its effectiveness 
include settings for
players coagulation factors count
script run time interval
and bandage type applied 

ADD- Saline Flush Prefilled 10ml Syringe as a item for clearing iv

**Comment by Th3Jock3R** [comment]
I would like to request the addition of a feature that enables blood volume to regenerate over time. It would be nice, if the feature include the option to adjust the regeneration rate via CBA settings (ml/hour). Ideally, there would be an option to require fluid consumption for regeneration

**Comment by TubaHorse** [comment]
I think a cool addition to available equipment would be an AutoPulse-like device. Some sort of equipment that you attach to a unit's chest that maintains CPR while you move them or otherwise work on them. It would make sense for it to take up a larger amount of inventory space to balance it out. Maybe it causes severe pain when used because it's a piston crushing your chest to keep your heart beating?

Now that I think about it... maybe ribs should be a breakable bone in the game? Susceptible to crushing damage?

**Comment by ItsGiantCookie** [comment]
A compression bag for IVs to increase flow rate with the downside of taking some time to prepare the bag would be a god addittion in my opinion

**Comment by Will-Nichols** [comment]
> I think a cool addition to available equipment would be an AutoPulse-like device. Some sort of equipment that you attach to a unit's chest that maintains CPR while you move them or otherwise work on them. It would make sense for it to take up a larger amount of inventory space to balance it out. Maybe it causes severe pain when used because it's a piston crushing your chest to keep your heart beating?
> 
> Now that I think about it... maybe ribs should be a breakable bone in the game? Susceptible to crushing damage?

It's been a while but I do remember a function in ACE (maybe a long time ago) that you could possibly crack/break ribs doing compressions in ACE Medical; or maybe that was a different mod that added something to ACE. I know it was one we ran in a previous milsim unit. If I can find the details I will post that in here.

**Comment by Will-Nichols** [comment]
> A compression bag for IVs to increase flow rate with the downside of taking some time to prepare the bag would be a god addittion in my opinion

Are you thinking about a "Pressure Infusion Bag"? If so they aren't really used for field operations because you have to maintain a steady state of pressure with the bag from an air source like an air compressor or hospital supplied air (which again comes from a big ass air compressor that distributes air throughout the hospital/surgery center.); I sell them for a living. 

In my humble opinion that's not something that would be practical; either in game or in real life if you are only using it in the field. Way too many milsim units want realism for medical but shortcuts on doing the actual medical stuff like surgical, broken bones and recovery/healing.

**Comment by Will-Nichols** [comment]
> > I think a cool addition to available equipment would be an AutoPulse-like device. Some sort of equipment that you attach to a unit's chest that maintains CPR while you move them or otherwise work on them. It would make sense for it to take up a larger amount of inventory space to balance it out. Maybe it causes severe pain when used because it's a piston crushing your chest to keep your heart beating?
> > Now that I think about it... maybe ribs should be a breakable bone in the game? Susceptible to crushing damage?
> 
> It's been a while but I do remember a function in ACE (maybe a long time ago) that you could possibly crack/break ribs doing compressions in ACE Medical; or maybe that was a different mod that added something to ACE. I know it was one we ran in a previous milsim unit. If I can find the details I will post that in here.

It wasn't ACE. It was XMed (https://github.com/X39/XMedSys1)

**Comment by ghost** [comment]
hey if its not too big of a deal, id love for the classnames file here on github to be updated with all the items not currently listed. im currently making a mission and im not 100% sure on how KAT medical fully works yet, so for the medics i want to give them an arsenal with all the medical items in it and i need all the classnames for that. the only problem is, theres nowhere i can find with all the classnames available :/

**Comment by ItsGiantCookie** [comment]
> > A compression bag for IVs to increase flow rate with the downside of taking some time to prepare the bag would be a god addittion in my opinion
> 
> Are you thinking about a "Pressure Infusion Bag"? If so they aren't really used for field operations because you have to maintain a steady state of pressure with the bag from an air source like an air compressor or hospital supplied air (which again comes from a big ass air compressor that distributes air throughout the hospital/surgery center.); I sell them for a living.
> 
> In my humble opinion that's not something that would be practical; either in game or in real life if you are only using it in the field. Way too many milsim units want realism for medical but shortcuts on doing the actual medical stuff like surgical, broken bones and recovery/healing.

I completely forgot I posted this but they are used (at least in the german army, source me and my former colleagues) in the TEC-Phase, sometimes in TFC if the need arises and especially in PFC (but that's mainly special and specialized forces). The design we use come with a hand-pump and pressure gauge built in, which means you don't need a continuous pressure source but they need to be checked frequently (which you do anyways if you do your job properly).

On the point of practicality, from what I gathered from the Medics and EMTs I worked with while in, who used them in action, it's pretty damn pratical, once you have a secured IV/IO access, you hook up 500mL to 1L of fluids, pump it up and work away on the rest of your algorithm. So to them it's a big time saver, especially when you work in pairs. 
It also is sometimes required for patients who are going into hypovolaemic or hemorrhagic shock, since you need to infuse up to 1000 ml of fluids in 10–15 minutes or 1 unit of blood in 20 minutes.

The WAKN Guidelines (S1-Leitlinie 001-042 (This one is for IO Infusions)), which are in part made by the Bundeswehr, also support the use of pressure infusion in pre-clinical and emergency cases.

**Comment by shawijawi-cellfvg43** [comment]
Could yall add basic medical ai that can apply chest seals, drain fluids,
Needle decompression, recovery position, and head turns?

**Comment by shawijawi-cellfvg43** [comment]
nvm I'll do it myself
